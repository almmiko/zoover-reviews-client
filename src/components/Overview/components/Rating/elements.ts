import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 100%;
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, .14);
  border-radius: 3px;
  position: relative;
  text-align: right;
  padding: 15px 40px 20px 15px;
  background-color: #fff;  
`;

export const Title = styled.div`
  color: #9e9e9e;
  font-size: 18px;
`;

export const Value = styled.div`
  font-size: 25px;
  color: #757575;
`;

export const IconBox = styled.div`
  position: absolute;
  top: -30px;
  left: 25px;
  padding: 30px;
  border-radius: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  background: linear-gradient(60deg, #66bb6a, #43a047);
  box-shadow: 0 12px 20px -10px rgba(76, 175, 80, .28), 
              0 4px 20px 0 rgba(0, 0, 0, .12),
              0 7px 8px -5px rgba(76, 175, 80, .2);
  
  i {
    font-size: 40px;
    color: #fff;
  }
`;
