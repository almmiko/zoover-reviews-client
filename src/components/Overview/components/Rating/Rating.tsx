import React from 'react';
import { Title, Value, Wrapper, IconBox } from './elements';
import { Icon } from 'antd';

type Props = {
  rating: number | undefined,
  loaded: boolean,
  apiError: boolean
}

class Rating extends React.Component<Props> {
  render() {
    const { loaded, rating, apiError } = this.props;
    return (
      <Wrapper>
        <IconBox>
          <Icon type="star" />
        </IconBox>
        <Title>Average Rating</Title>
        <Value error={apiError}>{apiError ? 'Error while fetching data' : (loaded ? rating : 'Loading...')}</Value>
      </Wrapper>
    );
  }
}

export default Rating;
