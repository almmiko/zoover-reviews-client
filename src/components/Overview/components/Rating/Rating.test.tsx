import { shallow } from 'enzyme';
import Rating from './Rating';
import React from 'react';

it('renders correctly', () => {
  const tree = shallow(<Rating loaded={true} apiError={false} rating={12} />);
  expect(tree).toMatchSnapshot();
});
