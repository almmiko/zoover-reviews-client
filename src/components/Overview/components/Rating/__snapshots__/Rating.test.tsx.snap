// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <Icon
      type="star"
    />
  </styled.div>
  <styled.div>
    Average Rating
  </styled.div>
  <styled.div
    error={false}
  >
    12
  </styled.div>
</styled.div>
`;
